api/src/main/java/org/apache/iceberg/catalog/ViewCatalog.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.catalog;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchViewException;
+import org.apache.iceberg.exceptions.NotFoundException;
+import org.apache.iceberg.view.View;
+import org.apache.iceberg.view.ViewRepresentation;
+
+/**
+ * A Catalog API for view create, drop, and load operations.
+ */
+public interface ViewCatalog {
+
+  /**
+   * Return the name for this catalog.
+   *
+   * @return this catalog's name
+   */
+  String name();
+
+  /**
+   * Return all the identifiers under this namespace.
+   *
+   * @param namespace a namespace
+   * @return a list of identifiers for views
+   * @throws NotFoundException if the namespace is not found
+   */
+  List<TableIdentifier> listViews(Namespace namespace);
+
+  /**
+   * Load a view.
+   *
+   * @param identifier a view identifier
+   * @return instance of {@link View} implementation referred by the identifier
+   * @throws NoSuchViewException if the view does not exist
+   */
+  View loadView(TableIdentifier identifier);
+
+  /**
+   * Check whether view exists.
+   *
+   * @param identifier a view identifier
+   * @return true if the table exists, false otherwise
+   */
+  default boolean viewExists(TableIdentifier identifier) {
+    try {
+      loadView(identifier);
+      return true;
+    } catch (NoSuchViewException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Create a view.
+   *
+   * @param identifier a view identifier
+   * @param representations a list of view representations
+   * @param properties a string map of view properties
+   */
+  View createView(
+      TableIdentifier identifier,
+      List<ViewRepresentation> representations,
```

Comment:
I find it a little odd that we're exposing the list of representations at this level without a wrapper class.  I just feel like there are a number of cases where we would want encapsulating data/information.  I would suggest adding a `ViewMetadata` as a container for this list.

Reply:
Having that `ViewMetadata` encapsulation seems to be against the structure defined in spec overall. Also I think we cannot use the term `ViewMetadata` since that is a term which means the entire view metadata including all versions in the spec: https://github.com/apache/iceberg/blob/master/format/view-spec.md?plain=1#L57

In my opinion, in the long term what we need is a builder here like `Catalog.buildTable(...)`, so something like 

```
ViewCatalog.buildView(TableIdentifier.of("db","table"))
  .addRepresentation(SqlViewRepresentation.of("select * from table"))
  .addProperty("key", "val")
  .build()
```

But for now this `createView` method should be sufficient

Reply:
Yeah ViewMetadata is reserved for the overall metadata for the view. The list of view reprsentations is part of the "versions" in the view.

Reply:
Also I think we'll need to pass in a Schema @jzhuge ?

Reply:
> But for now this createView method should be sufficient

I think we should create a builder now so that we can see the whole proposed API, not just part of it. We took the approach of updating `createTable` for different cases for a long time and needed to introduce the builder API because it was getting so annoying to maintain all the methods.

This is missing critical parts of the view spec right now, like schema, and I don't think we can move forward until those are represented.

```java
catalog.buildView(identifier, schema)
    .withQuery(selectStatement, dialect)
    .withProperty("key", "value")
    .withProperties(props)
    .withLocation(s3Location)
    .create();
```

Reply:
Adding the buildView API.